server/server-proxy/src/main/java/software/amazon/smithy/java/server/ProxyService.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.server;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiFunction;
+import software.amazon.smithy.java.auth.api.identity.Identity;
+import software.amazon.smithy.java.aws.client.core.settings.RegionSetting;
+import software.amazon.smithy.java.client.core.auth.identity.IdentityResolver;
+import software.amazon.smithy.java.client.core.auth.scheme.AuthSchemeResolver;
+import software.amazon.smithy.java.client.core.endpoint.Endpoint;
+import software.amazon.smithy.java.core.error.ModeledException;
+import software.amazon.smithy.java.core.schema.Schema;
+import software.amazon.smithy.java.core.schema.SerializableStruct;
+import software.amazon.smithy.java.core.serde.TypeRegistry;
+import software.amazon.smithy.java.core.serde.document.Document;
+import software.amazon.smithy.java.dynamicclient.DocumentException;
+import software.amazon.smithy.java.dynamicclient.DynamicClient;
+import software.amazon.smithy.java.dynamicclient.DynamicOperation;
+import software.amazon.smithy.java.dynamicschemas.SchemaConverter;
+import software.amazon.smithy.java.dynamicschemas.WrappedDocument;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.TopDownIndex;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.shapes.ShapeType;
+import software.amazon.smithy.model.shapes.ToShapeId;
+
+public class ProxyService implements Service {
+
+    private final DynamicClient dynamicClient;
+    private final SchemaConverter schemaConverter;
+    private final Map<String, Operation<WrappedDocument, WrappedDocument>> operations;
+    private final TypeRegistry serviceErrorRegistry;
+    private final Model model;
+    private final ServiceShape service;
+    private final List<Operation<? extends SerializableStruct, ? extends SerializableStruct>> allOperations;
+    private final Schema schema;
+
+    private ProxyService(Builder builder, ServiceShape service, Model model) {
+        this.model = model;
+        this.service = service;
+        DynamicClient.Builder clientBuilder = DynamicClient.builder()
+                .service(service.getId())
+                .model(model)
+                .endpointResolver(params -> CompletableFuture
```

Comment:
this can use the staticEndpoint helper method: https://github.com/smithy-lang/smithy-java/blob/main/client/client-core/src/main/java/software/amazon/smithy/java/client/core/endpoint/EndpointResolver.java#L49